### ISSUE TITLE
Question about struct allocation for Disruptor

### ISSUE THREAD

**Issue by DzmitryShylovich** [RuedigerMoeller/fast-serialization#105]
In my app I use [LMAX Disruptor](https://github.com/LMAX-Exchange/disruptor) to collect events and later send them via TCP. Here's my event object:

``` java
public class ReplicationEvent {
    protected long sequence;
    protected byte[] data;
    protected long length;
}
```

I think to make this object a struct will be very beneficial as I don't need to encode/decode it.

Disruptor pre-allocates memory for events to achieve 0 gc impact. So I need to provide a factory for my event objects. What is the best way to implement such factory?

My current solution looks like:

``` java
public class ReplicationEventFactory implements EventFactory<ReplicationEvent> {

    private FSTStructAllocator allocator;
    private final ReplicationEvent template = new ReplicationEvent();

    public ReplicationEventFactory(final int size) {
        Assert.isTrue(size >= 0, "Size must not be negative");
        initStructFactory();
        this.allocator = new FSTStructAllocator(size);
    }

    @Override
    public ReplicationEvent newInstance() {
        return allocator.newStruct(template).detach();
    }

    private void initStructFactory() {
        FSTStructFactory.getInstance().registerClz(ReplicationEvent.class);
    }

}
```

Should I call `allocator.newStruct` every time or it will be better to allocate an array of objects and then return 1 by 1?
Should I `detach` every instance?

Thank you.

**Comment by RuedigerMoeller** [comment]
If you pass in a decent size in the Allocator Constructor, fst will allocate many struct instances from a single chunk of memory, so "newStruct" is basically bumping a pointer.

If you want to keep the "pointer" you actually need to detach() (=an allocation). Usually one avoids detaching by reusing a single structpointer instance and move its offset and address to the ReplicationEvent being processed.
